split-index.c

```diff
@@ -146,8 +146,6 @@ void merge_base_index(struct index_state *istate)
 	si->nr_replacements = 0;
 	ewah_each_bit(si->replace_bitmap, replace_entry, istate);
 	ewah_each_bit(si->delete_bitmap, mark_entry_for_delete, istate);
-	if (si->nr_deletions)
-		remove_marked_cache_entries(istate);
```

Comment:
This patch *might* be correct.

It is the duty of the commit message to convince the reader of that, and it fails rather miserably.

Instead, it forces the reader to re-think the entire thought process that must have gone into crafting this commit *again*, which is rather unfortunate (and highly discouraged in the Git project).

Now, when *I* read this, I cannot help but think that the patch is wrong, especially when I see that the `si` refers to the `split_index` field of the index structure that was passed into this function. You see, the split index *can* delete index entries. And what other way would we have to fold those changes into the base index than via this code here?

In other words, I am now (after spending an unfortunate amount of time due to the too-short commit message) rather strongly convinced that this commit would introduce a serious regression in the split index feature.